File: baselibrary/src/main/java/com/aslan/baselibrary/http/error/HttpErrorFunctionCompletable.java
package com.aslan.baselibrary.http.error;

import android.content.Context;
import io.reactivex.Completable;

/**
 * 处理http异常
 *
 * @author Aslan chenhengfei@yy.com
 * @date 2020/6/10
 */
public class HttpErrorFunctionCompletable extends BaseHttpErrorFunction<Completable> {

  public HttpErrorFunctionCompletable(Context context) {
    super(context);
  }

  @Override
  public Completable error(Exception ex) {
    return Completable.error(ex);
  }
}

File: baselibrary/src/main/java/com/aslan/baselibrary/http/error/HttpErrorFunctionCompletable.kt
package com.aslan.baselibrary.http.error

import android.content.Context
import io.reactivex.Completable

/**
 * 处理http异常
 *
 * @author Aslan chenhengfei@yy.com
 * @date 2020/6/10
 */
open class HttpErrorFunctionCompletable(context: Context) :
    BaseHttpErrorFunction<Completable>(context) {
    override fun error(ex: Exception): Completable {
        return Completable.error(ex)
    }
}